src/promise.rs

```diff
@@ -42,13 +50,17 @@ impl<T> Sender<T> {
 /// ```
 ///
 /// If you enable the `tokio` feature you can use `poll-promise` with the [tokio](https://github.com/tokio-rs/tokio)
-/// runtime to run `async` tasks using [`Promise::spawn_async`] and [`Promise::spawn_blocking`].
+/// runtime to run `async` tasks using [`Promise::spawn_async`], [`Promise::spawn_local`], and [`Promise::spawn_blocking`].
 #[must_use]
 pub struct Promise<T: Send + 'static> {
     data: PromiseImpl<T>,
+    task_type: TaskType,
 
     #[cfg(feature = "tokio")]
     join_handle: Option<tokio::task::JoinHandle<()>>,
+
+    #[cfg(feature = "smol")]
+    task: Option<smol::Task<()>>,
```

Comment:
let's call it `smol_task` for clarity

```suggestion
    smol_task: Option<smol::Task<()>>,
```